### PR TITLE
Completed train

### PR DESCRIPTION
**Pull request by MMDPROJECT**
`train.py` is completed. It has the following capabilities:
- Initialize a custom **Atari** environment.
- Train the agent in `agent.py` in a train loop for a certain number of iterations.
- Log the results and displaying them after a few episodes. It also creates a plot of `loss` and `reward` histories for each episode.

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+# parameters of the training loop 
+total_interactions = 0 # total number of the interactions, that the agent had so far (each stack of the frames is counted once).
+
+
+# logging variables (accumulated over all episodes)
+history_of_total_losses = []
+history_of_total_rewards = []
+episode_cnt = 0
+num_of_last_episodes_to_avg = 1000
+log_display_step = 100000
```
Comment (Retriever5500) [inline train.py:64]: Lower `num_of_last_episodes_to_avg` to something lower, maybe 100 or 200
also lower `log_display_step`, something about 10000

File: train.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        obs = next_obs
+
+        # logging (accumlated over each episode)
+        total_interactions += 1
+        episode_finished = terminated or truncated
+        episode_total_loss += loss
+        episode_total_reward += reward
+
+        # display logs every log_display_step
+        if (total_interactions % log_display_step) == 0 and (total_interactions > 0) and (episode_cnt >= num_of_last_episodes_to_avg):
```
Comment (Retriever5500) [inline train.py:97]: Save the model throughout the training, so we don't have to wait for the training to finish to test it.

**Review by Retriever5500 — CHANGES_REQUESTED** [review]
